### ISSUE TITLE
getJSDocComment method won't pick up the comment on the object expression

### ISSUE THREAD

**Issue by fa93hws** [eslint/eslint#11898]
**Tell us about your environment**
Not important, should be able to reproduce on all platforms.

* **ESLint Version: 6.0.1**
* **Node Version: 10.16.0**
* **npm Version: 6.9.0**

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
empty (not related to this bug)

<details>
<summary>Configuration</summary>

```js

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**
The JSDoc should be picked up by calling `getJSDocComment` on object expression

```js
/** docs */
const a = {};
```

**What did you expect to happen?**
JSDocComment on object expression is picked up

**What actually happened? Please include the actual, raw output from ESLint.**
The function return null

**Are you willing to submit a pull request to fix this bug?**
https://github.com/eslint/eslint/pull/11897
